lib/dataset/bin.cpp

```diff
@@ -294,11 +294,17 @@ class TargetBinBuilder {
             allsorted(bin_coords.at(dim), dim)) {
           const auto &bin_coord = bin_coords.at(dim);
           const bool histogram =
-              bin_coord.dims()[dim] == indices.dims()[dim] + 1;
-          const auto begin =
+              bin_coord.dims()[dim] ==
+              (indices.dims().contains(dim) ? indices.dims()[dim] : 1) + 1;
+          auto begin =
               begin_edge(histogram ? left_edge(bin_coord) : bin_coord, key);
-          const auto end = histogram ? end_edge(right_edge(bin_coord), key)
-                                     : begin + 2 * units::none;
+          auto end = histogram ? end_edge(right_edge(bin_coord), key)
+                               : begin + 2 * units::none;
+          // When we have bin edges (of length 2) for a dimension that is not
+          // a dimension of the input it needs to be squeezed to avoid problems
+          // in various places later on.
+          begin = squeeze(begin, std::nullopt);
+          end = squeeze(end, std::nullopt);
```

Comment:
Can this cause problems when squeezing a dimension other than `dim`?

Reply:
I don't think so because `transform` will handle this equivalently. But given the overall code complexity and number of possible cases I cannot really guarantee that 100%.

Reply:
Can you add a test with a length=1 dimension but bin in a longer one?